### PR TITLE
Use ListItem component to render items in List component.

### LINKED ISSUE #1: 1. As a user, I want to read the names of all of the items in my shopping list
## Summary

The most important feature of our app is the ability to look at what's on the list, so the user knows what they need to buy. The `List` component is the best place in our code to do that. In fact, this feature is already partially implemented. Refer to the acceptance criteria in this issue, and the existing code, to understand the work that remains.

The `List` component is rendered on the `/list` page. Because the navigation links are currently broken, you'll have to type that path into your browser's address bar.

## Acceptance criteria
- [x] The `streamListItems` function is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [x] #17

## Notes
- ⚠️ Your **only goal** is to make sure that the names of the items render as expected. You might be tempted to modify `ListItem.jsx` to show more information – you will do that **later**. If you want a peek at the future, [this wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png) is what your list could look like closer to the end of the project.
- The data you need is already being passed into the `List` component. What steps do you need to take in order to render it?

### PR DESCRIPTION
**Pull request by AslanTaheri**
## Description

Data is received from the Firebase database via the `App` component.
In the `List` component, we are mapping the data using the `ListItem` component. The `name` attribute is de-structured from the data in the `ListItem` component. Using the `id` attribute of each passed on item, we generated a `key` for every list item.

This code enables the user to view the items that exist on their shopping list.
We made the decision that the `map()` method would be the most efficient way of dynamically rendering the instances of the `ListItem` component via the `List` component.

## Related Issue

closes #1 

## Acceptance Criteria

 

- [x] In List.jsx, the ListItem component is used to render the name of each item

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
image

### After
image

## Testing Steps / QA Criteria

Click the list link on the navbar or manually append `/list` to the home page URL to view the list of items.

### CODE REVIEW

**Review by shangguanwang — APPROVED** [review]
Well done! I like how you created the sample data to test this feature!

**Review by rdpfeifle — APPROVED** [review]
It looks really good, great job! I'd only suggest deleting the comment in the List.jsx.

**Review by shangguanwang — APPROVED** [review]

File: src/views/List.jsx [hunk 1]
```diff
@@ -7,11 +7,9 @@ export function List({ data }) {
 				Hello from the <code>/list</code> page!
 			</p>
 			<ul>
-				{/**
-				 * TODO: write some JavaScript that renders the `data` array
-				 * using the `ListItem` component that's imported at the top
-				 * of this file.
-				 */}
+				{data.map((item) => (
+					<ListItem key={item.id} name={item.name} />
```
Comment (mentalcaries) [inline src/views/List.jsx:11]: Nice work with including that key!

**Review by cafloyd — APPROVED** [review]
Great work @AslanTaheri @CorinaMurg !! Really appreciate the thorough description and the way you talked through your decisions!

**Review by mentalcaries — APPROVED** [review]
Well done! Great write up for the PR!

File: src/views/List.jsx (hunk 1 above)
Comment (AslanTaheri) [reply src/views/List.jsx:11]: Thank you!

**Review by rdpfeifle — APPROVED** [review]

**Pull request merged**